docs/process/integrating-with-fxa.md

```diff
@@ -4,28 +4,123 @@ title: Integration with FxA
+3. Confirm returned state token by comparing it with state token associated with the local session.
+4. Exchange code for an access token and possibly a refresh token (for clients that request offline use).
+5. Fetch user profile information with access token from the FxA Profile Server.
+6. Associate profile information with local session and possibly create an account.
+
+### OAuth Credentials
+
+1. client_id - a public identifier that is used to identify your service. Can be public.
+2. client_secret - a private secret that is sent from the backend when interacting with the OAuth server. Must not be shared publicly, checked into a public repository, or bundled with compiled code.
+
+### Self Hosted Email-first Flow
+
+1. Initialize top of funnel metrics by calling [/metrics-flow request][metrics-flow-request] with the required query parameters:
+   a. entrypoint
+   b. form_type (must be the string ‘email’)
+   c. utm_source
+   d. utm_campaign
+2. Propagate email, flow_id and flow_begin_time query parameters to /authentication request
+
+To test without CORS errors using https://stable.dev.lcip.org/, your test application must have one of the following URLs:
+
+- http://127.0.0.1:8001
+- http://localhost:8000
+- http://127.0.0.1:8000
+- https://www.mozilla.org
+- https://www.allizom.org
+- https://www-demo5.allizom.org
+- https://www-demo4.allizom.org
+- https://www-demo3.allizom.org
+- https://www-dev.allizom.org
+
+### /authorization query parameters
+
+1. client_id (required)
+2. [scope](#scopes) (required)
+3. state (required)
+4. entrypoint (required)
+5. email (required for [self hosted email-first flow](#self-hosted-email-first-flow))
+6. flow_begin_time (required for [self hosted email-first flow](#self-hosted-email-first-flow))
+7. flow_id (required for [self hosted email-first flow](#self-hosted-email-first-flow))
+8. code_challenge (required for PKCE)
+9. code_challenge_method (required for PKCE)
+10. action (suggested, should be the string ‘email’)
+11. access_type (suggested)
+12. utm_campaign (suggested)
+13. utm_source (suggested)
+14. utm_medium (optional)
+15. utm_term (optional)
+
+### Scopes
+
+1. Sync data
+2. Profile data
+
+### User Data Hygiene / GDPR
+
+1. Accounts should use uid rather than email address as the primary key. An account’s primary email address can change.
+2. [Primary email changed notifications](https://github.com/mozilla/fxa-auth-server/blob/master/docs/service_notifications.md#change-of-primary-email-address-event) should update the contact email stored with the account.
+3. If profile information is stored, listen for [profile changed notifications](https://github.com/mozilla/fxa-auth-server/blob/master/docs/service_notifications.md#change-of-profile-data) or periodically refresh the profile information by using refresh token to create a fresh access token that can fetch profile information.
```

Comment:
Should new RPs expect to receive these via webhooks, or not yet?

Reply:
Yesterday a PR adding the remaining needed events to the event-broker landed, so I think all new RP's should use the webhook integration. It's also fully documented with code examples.